apps/dashboard/app/controllers/module_browser_controller.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+class ModuleBrowserController < ApplicationController
+  def index
+    # FIXME: select clusters from Configuration.job_clusters
+    @clusters = OodAppkit.clusters.select(&:job_allow?).reject { |c| c.metadata.hidden }
+    @selected_clusters = params[:cluster]
+    @modules = @clusters.flat_map do |cluster|
+      HpcModule.all(cluster.id).each { |mod| mod.cluster = cluster.id }
```

Comment:
Looks like we need to refactor `HpcModule#all` to actually return all modules.

Maybe it's an array and we can use the cluster attribute in another layer to group by.